calliope/backend/pyomo/util.py

```diff
@@ -199,3 +201,22 @@ def loc_tech_is_in(backend_model, loc_tech, model_set):
         return True
     else:
         return False
+
+
+def get_domain(var: xr.DataArray) -> str:
+    def check_sign(var):
+        if "cost" in var.name or var.name in ["resource", "loc_coordinates"]:
+            return ""
+        else:
+            return "NonNegative"
+
+    if var.dtype == bool:
```

Comment:
I'm guessing this is the same as before, although probably it's safe for booleans.  Best to be consistent though:
```suggestion
    if var.dtype.kind == "b":
```